python_modules/dagster/dagster/_core/definitions/asset_graph.py

```diff
@@ -197,6 +206,25 @@ def from_assets(
                 backfill_policies_by_key.update({key: asset.backfill_policy for key in asset.keys})
                 code_versions_by_key.update(asset.code_versions_by_key)
 
+                is_observable = asset.execution_type == AssetExecutionType.OBSERVATION
+                is_observable_by_key.update({key: is_observable for key in asset.keys})
+
+                # Set auto_observe_interval_minutes for external observable assets
+                # This can be removed when/if we have a a solution for mapping
+                # `auto_observe_interval_minutes` to an AutoMaterialzePolicy
+                first_key = next(iter(asset.keys), None)
+                if (
+                    first_key
+                    and SYSTEM_METADATA_KEY_AUTO_OBSERVE_INTERVAL_MINUTES
+                    in asset.metadata_by_key[first_key]
+                ):
+                    interval = asset.metadata_by_key[first_key][
+                        SYSTEM_METADATA_KEY_AUTO_OBSERVE_INTERVAL_MINUTES
+                    ]
+                    auto_observe_interval_minutes_by_key.update(
+                        {key: interval for key in asset.keys}
+                    )
+
```

Comment:
There is a separate part of the code that already sets `auto_observe_interval_minutes` for `SourceAsset`. This sets it for observable `AssetsDefinition`.